### PR TITLE
[PYTHON] Issue 8075 - additionalProperties support

### LINKED ISSUE #8075: [PYTHON] additionalProperties not supported
##### Description

Generated Python clients do not have the ability to access the additional properties of models with `additionalProperties` defined. The only ones that can be accessed are those that are explicitly defined in the model and the rest are ignored. In contrast, the generated Java models extend HashMap when `additionalProperties` are defined and they can get accessed by Map#get(String).

##### Swagger-codegen version

2.3.1

##### Swagger declaration file content or url

```yaml
  Element:
    type: "object"
    required:
    - "id"
    properties:
      id:
        type: "string"
      name:
        type: "string"
    additionalProperties:
      type: "object"
```
Full specification: https://github.com/Open-MBEE/mms/blob/develop/mms-ent/repo-amp/src/main/amp/web/mms/mms.swagger.yaml

##### Command line used for generation

```shell
java -jar ~/Downloads/swagger-codegen-cli-2.3.1.jar generate -i ~/Documents/mms.swagger.yaml -l php -o ~/tmp
```

##### Steps to reproduce

Inspecting the generated `Element.py` file after executing the command above with the specification above. There is also a usage of `additionalProperties` in the Petstore API that exhibits the same behavior.

##### Related issues/PRs

https://github.com/swagger-api/swagger-codegen/pull/6130

##### Suggest a fix/enhancement

I believe that the additional properties can be stored in a dict, analogously to how they are handled in Java.

### PR DESCRIPTION
**Pull request by ivan-gomes**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [X] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Fixes #8075.

### CODE REVIEW

**Comment by ivan-gomes** [comment]
@kenjones-cisco

File: modules/swagger-codegen/src/main/resources/python/api_client.mustache [hunk 1]
```diff
@@ -620,6 +620,12 @@ class ApiClient(object):
```
Comment (kenjones-cisco) [inline modules/swagger-codegen/src/main/resources/python/api_client.mustache]: file permissions should stay at `0644`

File: modules/swagger-codegen/src/main/resources/python/model.mustache [hunk 2]
```diff
@@ -14,7 +14,7 @@ import six
```
Comment (kenjones-cisco) [inline modules/swagger-codegen/src/main/resources/python/model.mustache]: file permissions should stay at `0644`

**Comment by kenjones-cisco** [comment]
Minor revert of the file permissions. Otherwise it LGTM.

Thanks! @ivan-gomes

**Comment by ivan-gomes** [comment]
Reverted the file permissions. Good catch! Thanks.

**Comment by HugoMario** [comment]
after reviewing and test locally (merging upstream master into `Open-MBEE:issue-8075` locally) everything looked good to me, so going to merge this.

Thanks a lot @ivan-gomes !!!

**Comment by HugoMario** [comment]
it seems my verification was not ok since build is failing for python, anyway i'll check if it's something i can fix, otherwise i'll file a new issue.

**Pull request merged**